rvw/parser.js

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env node
+
+'use strict';
+const Parser = require('../.lib/parser.js');
+
+/**
+ * Recognizes the accesses to the platform Rivisteweb
+ * @param  {Object} parsedUrl an object representing the URL to analyze
+ *                            main attributes: pathname, query, hostname
+ * @param  {Object} ec        an object representing the EC whose URL is being analyzed
+ * @return {Object} the result
+ */
+module.exports = new Parser(function analyseEC(parsedUrl, ec) {
+  let result = {};
+  let path   = parsedUrl.pathname;
+  // uncomment this line if you need parameters
+  // let param = parsedUrl.query || {};
+
+  // use console.error for debuging
+  // console.error(parsedUrl);
+
+  let match;
+
+  if ((match = /^\/download\/article\/([0-9]+.[0-9]+\/[0-9]+)$/i.exec(path)) !== null) {
+    // https://www.rivisteweb.it/download/article/10.7390/103852
+    result.rtype    = 'ARTICLE';
+    result.mime     = 'PDF';
+    result.unitid = match[1];
+    result.doi = match[1];
+
+  } else if ((match = /^\/doi\/([0-9]+.[0-9]+\/[0-9]+)$/i.exec(path)) !== null) {
+    // https://www.rivisteweb.it/doi/10.7373/82132
+    result.rtype    = 'ARTICLE';
+    result.mime     = 'HTML';
+    result.unitid   = match[1];
+    result.doi = match[1];
+  } else if ((match = /^\/rivisteweb\/searchresults$/i.exec(path)) !== null) {
```

Comment:
`RegExp.exec()` is not usefull since we're not using `match` (the result). You can use `RegExp.test()`.

```suggestion
  } else if (/^\/rivisteweb\/searchresults$/i.test(path)) {
```

> **Note**
> Because it's a very small change that doesn't add/remove a feature/bug, I merged the PR and did it in e3d4bd6